pilot/pkg/kube/inject/app_probe.go

```diff
@@ -74,30 +77,41 @@ func rewriteAppHTTPProbe(spec *SidecarInjectionSpec, podSpec *corev1.PodSpec) {
 				// Matches the regex pattern, but without actual values provided.
 				if len(c.Args) <= i+1 {
 					log.Errorf("No statusPort value provided, skip app probe rewriting")
-					return
+					return -1
 				}
 				portStr = c.Args[i+1]
 			}
 			p, err := strconv.Atoi(portStr)
 			if err != nil {
 				log.Errorf("Failed to convert statusPort to int %v, err %v", portStr, err)
-				return
+				return -1
 			}
 			statusPort = p
 			break
 		}
 	}
+	return statusPort
+}
+
+func calculateProbeRewrite(podSpec *corev1.PodSpec, spec *SidecarInjectionSpec, modify bool) []rfc6902PatchOperation {
```

Comment:
nit: naming .... calculate? Maybe something like `createProbeRewritePatch` would be more appropriate?

Reply:
Yup! I already have a `createProbeRewritePatch` below, which invoke this function with `modify=false`.